server/py/framework/utils/clients/messaging.py

```diff
@@ -282,3 +289,20 @@ async def _resolve_request_kwargs_from_request(
 
         request_kwargs.update(**kwargs)
         return request_kwargs
+
+    @staticmethod
+    def _get_prefix_and_version(path: str):
+        match = PREFIX_GROUPING.match(path)
+        prefix = match.group(1)
+        version = match.group(2) or "v1"
+        return prefix, version
+
+    def _prepare_request_data(self, method: str, path: str):
+        prefix, version = self._get_prefix_and_version(path)
+
+        # Remove the service and version prefix from the path
+        # The service prefix is to be replaced with the new service name
+        # The version will be re-added or default to v1 if not present
```

Comment:
This comment should be part of  `_get_prefix_and_version`

Reply:
@rokatyy Actually, it is related to the `proxy_request` method, so I moved it there